File: CyberSheep.java
import javax.swing.*;
import java.util.Random;

public class CyberSheep extends Animal{
    CyberSheep(World world1, int x, int y) {
        SetStrength(11);
        SetInitiative(4);
        SetCords(x, y);
        world = world1;
    }

    @Override
    public String getName() {
        return "CYBER-SHEEP";
    }
    @Override
    public void Drawing() {
        ImageIcon icon = new ImageIcon("src/Icons/cyber.png");
        world.GetButton(GetCords()).setIcon(icon);
    }
    @Override
    public void Action(){
        //GETTING DATA
        Polozenie tmp_cords = GetCords();
        Polozenie hogweed_coords= world.GetOrganisms().FindHogweed(tmp_cords);


        boolean move = false;

        while (!move) {
            if(hogweed_coords.x!=0){
                if (hogweed_coords.y < tmp_cords.y) {
                    SetCords(tmp_cords.x, tmp_cords.y - 1);
                    SetLastMove(Stale.UP);
                    move = true;
                }
                else if (hogweed_coords.y > tmp_cords.y) {
                    SetCords(tmp_cords.x, tmp_cords.y + 1);
                    SetLastMove(Stale.DOWN);
                    move = true;
                }
                else if (hogweed_coords.x < tmp_cords.x) {
                    SetCords(tmp_cords.x - 1, tmp_cords.y);
                    SetLastMove(Stale.LEFT);
                    move = true;
                }
                else if (hogweed_coords.x > tmp_cords.x) {
                    SetCords(tmp_cords.x + 1, tmp_cords.y);
                    SetLastMove(Stale.RIGHT);
                    move = true;
                }
            }
            else{
                Random random = new Random();
                int number = random.nextInt(4);

                //UP
                if (number == 0) {
                    if (tmp_cords.y > 1) {
                        SetCords(tmp_cords.x, tmp_cords.y - 1);
                        SetLastMove(Stale.UP);
                        move = true;
                    }
                    else continue;
                }
                //DOWN
                else if (number == 1) {
                    if (tmp_cords.y < world.GetHeight()) {
                        SetCords(tmp_cords.x, tmp_cords.y + 1);
                        SetLastMove(Stale.DOWN);
                        move = true;
                    }
                    else continue;
                }
                //RIGHT
                else if (number == 2) {
                    if (tmp_cords.x < world.GetWidth()) {
                        SetCords(tmp_cords.x + 1,tmp_cords.y);
                        SetLastMove(Stale.RIGHT);
                        move = true;
                    }
                    else continue;
                }
                //LEFT
                else {
                    if (tmp_cords.x > 1) {
                        SetCords(tmp_cords.x - 1, tmp_cords.y);
                        SetLastMove(Stale.LEFT);
                        move = true;
                    }
                    else continue;
                }
            }
        }
    }
}

File: cybersheep.py
import random
from animal import Animal
from Stale import Stale

class CyberSheep(Animal):
    def __init__(self, world1, x, y):
        super().__init__()
        self.SetStrength(11)
        self.SetInitiative(4)
        self.SetCords(x, y)
        self.world = world1

    def getName(self):
        return "CYBER-SHEEP"

    def Drawing(self):
        self.world.buttons[self.GetCords().y -1][self.GetCords().x -1].configure(bg='#0000FF')
    
    def New(self, text, reference, x, y):
        self.world.GetOrganisms().add(CyberSheep(reference, x, y))    


    def Action(self):
        # GETTING DATA
        tmp_cords = self.GetCords()
        hogweed_coords = self.world.GetOrganisms().FindHogweed(tmp_cords)

        move = False

        while not move:
            if hogweed_coords.x != 0:
                if hogweed_coords.y < tmp_cords.y:
                    self.SetCords(tmp_cords.x, tmp_cords.y - 1)
                    self.SetLastMove(Stale.UP)
                    move = True
                elif hogweed_coords.y > tmp_cords.y:
                    self.SetCords(tmp_cords.x, tmp_cords.y + 1)
                    self.SetLastMove(Stale.DOWN)
                    move = True
                elif hogweed_coords.x < tmp_cords.x:
                    self.SetCords(tmp_cords.x - 1, tmp_cords.y)
                    self.SetLastMove(Stale.LEFT)
                    move = True
                elif hogweed_coords.x > tmp_cords.x:
                    self.SetCords(tmp_cords.x + 1, tmp_cords.y)
                    self.SetLastMove(Stale.RIGHT)
                    move = True
            else:
                number = random.randint(0, 3)

                # UP
                if number == 0:
                    if tmp_cords.y > 1:
                        self.SetCords(tmp_cords.x, tmp_cords.y - 1)
                        self.SetLastMove(Stale.UP)
                        move = True
                    else:
                        continue
                # DOWN
                elif number == 1:
                    if tmp_cords.y < self.world.GetHeight():
                        self.SetCords(tmp_cords.x, tmp_cords.y + 1)
                        self.SetLastMove(Stale.DOWN)
                        move = True
                    else:
                        continue
                # RIGHT
                elif number == 2:
                    if tmp_cords.x < self.world.GetWidth():
                        self.SetCords(tmp_cords.x + 1, tmp_cords.y)
                        self.SetLastMove(Stale.RIGHT)
                        move = True
                    else:
                        continue
                # LEFT
                else:
                    if tmp_cords.x > 1:
                        self.SetCords(tmp_cords.x - 1, tmp_cords.y)
                        self.SetLastMove(Stale.LEFT)
                        move = True
                    else:
                        continue
